src/metabase/events/view_log.clj

```diff
@@ -12,16 +13,42 @@
    [steffan-westcott.clj-otel.api.trace.span :as span]
    [toucan2.core :as t2]))
 
-(defn increment-view-counts!
-  "Increments the view_count column for a model given a list of ids.
-   Assumes the model has one primary key `id`, and the column for the view count is named `view_count`"
-  [_model & _ids]
-  ;; Disable view_count updates to handle perf issues  (for now) (#44359)
-  #_
-  (when (seq ids)
-    (t2/query {:update (t2/table-name model)
-               :set    {:view_count [:+ :view_count [:inline 1]]}
-               :where  [:in :id ids]})))
+(defn- group-by-frequency
+  [ids]
+  (as-> ids items
+    (seq (frequencies items))
+    (group-by second items)
+    (update-vals items #(map first %))))
+
+(defn- increment-view-counts!*
+  [items]
+  (log/debugf "Increment view counts of %d items" (count items))
+  (try
+    (let [model->ids (as-> items items
+                       (filter (comp some? :id) items)
+                       (group-by :model items)
+                       (update-vals items #(map :id %)))]
+      (doseq [[model ids] model->ids]
+        (doseq [[cnt ids] (group-by-frequency ids)]
+          (log/debugf "Incrementing view count of %d and %s" (count ids) (pr-str (t2/select model :id [:in ids])))
+          (t2/query {:update (t2/table-name model)
+                     :set    {:view_count [:+ :view_count [:inline cnt]]}
+                     :where  [:in :id ids]}))))
+    (catch Exception e
+      (log/error e "Failed to increment view counts"))))
+
+(def ^{:private true
+       :once    true}
```

Comment:
IDK if you meant `defonce` here, but `^:once` is something different: https://ask.clojure.org/index.php/13050/what-does-once-meta-applied-for-anon-function-mean

Reply:
This should probably also be a delay. We don't need to start a grouper runner in AOT

Reply:
Updated to use `defonce` and also use delay.